eth_verify/rust/native/src/main.rs

```diff
@@ -0,0 +1,73 @@
+use alloy_primitives::FixedBytes;
+use eth_verify::{calculate_mpt_root, verify_block_hash, Header};
+use serde::{self, Deserialize};
+use serde_json;
+use std::error::Error;
+use std::fs::File;
+use std::io::BufReader;
+use tokio;
+use reqwest::Client;
+use std::env;
+use dotenv;
+
+#[derive(Deserialize)]
+struct BlockInfo {
+    number: u64,
+    hash: String,
+}
+
+#[tokio::main]
+async fn main() {
+    // Read JSON file from disk
+    let file = File::open("../../../inputs/block.json").expect("Failed to open file");
+    let reader = BufReader::new(file);
+    let block_info: BlockInfo = serde_json::from_reader(reader).expect("Failed to parse JSON");
+
+    println!("Block number: {}", block_info.number);
+    println!("Block hash: {}", block_info.hash);
+    // Fetch the latest block
+    // Load environment variables from .env file
+    dotenv::dotenv().expect("Failed to read .env file");
+    // Get the URL from the environment variable
+    let url = env::var("ETH_NODE_URL").expect("ETH_NODE_URL not set in .env file");
+    let resp = get_block_resp(block_info.number, &url)
+        .await
+        .expect("Failed to fetch block data");
+    // Deserialize the response to get block and transaction data
+    let block = serde_json::from_value(resp["result"].clone()).unwrap();
+    let result = calculate_mpt_root(block).await;
+    println!("Tx root generated and matched successfully: {}", result);
+    if !result {
+        println!("Tx root does not match");
+        return;
+    }
+
+    let header: Header = serde_json::from_value(resp["result"].clone()).unwrap();
```

Comment:
is the block same as header?

Reply:
They have a lot of fields in common so can deserialise what we need to create Header